London class 9_Turing - GayleNgozi_Thompson-Igwebike - HTML/CSS week 4

Gayle-Thompson-Igwebike opened the pull request:
working progress of resume website

Gayle-Thompson-Igwebike committed:
Current progress of resume

working progress of resume website

SallyMcGrath commented:
👋 Hi @GayleNThompsonIgwebike 

Please open your PR using the template provided and the title requested. Thanks!

https://www.awesomecodereviews.com/pull-request-template/

Gayle-Thompson-Igwebike committed:
updated containers and social media icons

I updated my containers and added my social media icons to my webpage.
I am still working on improving my implementing grids.....

Gayle-Thompson-Igwebike committed:
styling buttons and main text

SallyMcGrath commented:
(Thanks for updating!!)

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+}
+
+body{
+    /* display: grid; */
```
Whoops, some commented out code left in this PR. 

https://syllabus.codeyourfuture.io/guides/code-style-guide#dont-leave-lots-of-commented-out-code

SallyMcGrath replied:
```suggestion
```

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+
+body{
+    /* display: grid; */
+    /* grid-template-columns: ; */
```
```suggestion
```

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+body{
+    /* display: grid; */
+    /* grid-template-columns: ; */
+    /* min-width: auto; */
```
```suggestion
```

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+  width: 25px;
+}
+
+    /*
```
And here, let's clean this up. If you would like to share this code to discuss it, make a separate commit with this alternative stuff. You can check out different versions of the same page.

SallyMcGrath commented on index.html:
```diff
@@ -4,14 +4,81 @@
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
-    <!-- Add a link to your custom CSS here -->
+    <link rel="stylesheet" href="css/main.css">
 
```
I think right here might be a good place to load some custom web fonts. The fonts specified in the description are both Google Fonts, so you can grab them here: https://fonts.google.com/

What does MDN say about custom web fonts? https://developer.mozilla.org/en-US/docs/Learn/CSS/Styling_text/Web_fonts

SallyMcGrath commented on index.html:
```diff
@@ -4,14 +4,81 @@
+       <div class="container">
+         <div class="card">
+            <br>
+            <img src="/designs/resumepic.jpeg" alt="myphoto">
```
Is this alt text descriptive for someone who cannot see the picture? How would a screenreader read out the string "my-photo" and what is communicated by this?

https://supercooldesign.co.uk/blog/how-to-write-good-alt-text

SallyMcGrath commented on index.html:
```diff
@@ -4,14 +4,81 @@
+            <div class="blue-line">
+            <hr size="3px" width="50px" color="blue"> </div>
+
+            <div class="title"><p>SOFTWARE DEVELOPER</p></div> 
```
Is there a semantic html element you might use to describe a title or heading?

SallyMcGrath commented on index.html:
```diff
@@ -4,14 +4,81 @@
+            <button class="projects">PROJECTS</button>
+          </div>
+
+            <div class="socials-container"> 
```
What might this group of pictures be doing? What is their function? Is there a more meaningful way to describe this function? Should these link to places? Is this a list?

SallyMcGrath commented on index.html:
```diff
@@ -4,14 +4,81 @@
+            <div class="title"><p>SOFTWARE DEVELOPER</p></div> 
+
+            <div class="buttons">
+            <button class="resume">RESUME</button>
```
Should this be a button or a link styled as a button? What is the difference?

(Hint: links navigate you somewhere else in the DOM)

SallyMcGrath commented on index.html:
```diff
@@ -4,14 +4,81 @@
+           <p id=“greeting”><h1>Hello</h1></p>
+             <p id=“intro”>Here’s who I am & what I do</p>
+        <div class=“buttons2”>
+                    <button class="resume">RESUME</button>
```
Should this be a button or a link styled as a button?

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+     
+}
+
+#myLinks a {
```
In general, try to avoid the use of ids in your CSS. This isn't an absolute rule, but a helpful hint for managing specificity when your designs become more complex. Read around this idea: https://dev.to/clairecodes/reasons-not-to-use-ids-in-css-4ni4

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+  width:100%;
+  display: grid;
+  grid-template-areas: " left right";
+  position:absolute;
```
This looks pretty wild on my screen. In general, using position:absolute for layout like this creates more problems than it solves.

Try resizing your browser, and choosing different views in Devtools Device Mode. 

https://developer.chrome.com/docs/devtools/device-mode/

What happens to your layout? 😱 😂

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+    width: 50%;
+    grid-area: right;
+    margin: auto;
+    position: absolute;
```
I think this is all a lot more complicated than it needs to be, and I think the place it's going wrong is the grey box. Try removing the grey box and all this absolute positioning. 

Have a look into CSS linear-gradient. Here's a codepen you might find interesting. https://codepen.io/stefanjudis/pen/vYOjJdZ?editors=1100

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+  margin-left: 14px;
+}
+
+  .grey-box{
```
I don't think you need this and I think it's sending you down a rabbit hole of positioning nightmares. Try deleting this entirely and using a linear-gradient background on the body.

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+    display: flex;
+    flex-direction: column;
+    background-color: #F4ECE6;
+    right:-20em;
```
I think you can drop these negative positions here too. Work on the inside of the card component only.

Imagine the page as like a bento box. https://www.google.com/search?q=bento+box

Your card is like one delicious salad for your bento box. Your card can't know what kind of screen it will show up on, just like your salad doesn't know what bento box it will be spooned into. 🍱 

The page layout rules belong on the page level. Don't try to position each component by hand;  instead define a layout and "pour" each component into the layout section.

SallyMcGrath replied:
(This is a complicated but **very useful** concept. Don't feel dismayed if you don't get this straight away! Take this idea to study sessions and buddy calls and think about it together. )

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+img{
+    width: 120px;
+    height: 120px;
+    border-radius: 50%; 
```
nice circle!

Another great way to do this is with `clip-path`. Have a play https://bennettfeely.com/clippy/

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+.container{
+  width:100%;
+  display: grid;
+  grid-template-areas: " left right";
```
Here's a fun idea. You could  remove the gap rule and instead define _empty_  columns in CSS grid like this:

`grid-template-areas: ". card . intro .";`

Then you could try setting rules about how those columns are sized something like this:

`grid-template-columns: 1fr minmax(auto, 375px) 20px 3fr 1fr`

Have a play around.

SallyMcGrath commented on css/main.css:
```diff
@@ -1,3 +1,225 @@
+  padding: 120px 0 0 150px;
+  height: 32em;
+  right: -20em;
+  gap: px;
```
This CSS unit is missing a value

SallyMcGrath reviewed (commented):
Hi Gayle.

Thanks for your work so far. I can see you've put a lot of effort in. I've made a TON of comments here. Don't try to "fix" them all, certainly not at once.  I would take a look through the HTML comments first, and then come back and take another shot at the layout when you have a bit of space.

You're doing great, Gayle. Onward!

GayleNThompsonIgwebike commented:
Wow....thank you very much for this epic review of my work and your directions Sally. I truly appreciate the effort you must have put into this. I did not see this before now and just today, I deleted all my work to start over again because I just wasn't getting why it wont behave- so this is very very useful. THANK YOU. I will start it again with all your comments now as a guide. You are AWESOME Sally!!

Gayle-Thompson-Igwebike committed:
Started afresh post review.

Please ignore the massive commented code. Working progress

Gayle-Thompson-Igwebike committed:
Adding menu -draft-

Gayle-Thompson-Igwebike committed:
working on my mobile format

Gayle-Thompson-Igwebike committed:
work continues on project

Gayle-Thompson-Igwebike committed:
work in progress
